lib/GILGen/GILGenExpr.hpp

```diff
@@ -221,30 +285,26 @@ struct GILGenExpr : public ASTVisitor<GILGenExpr, gil::Value> {
     {
         using namespace glu::ast;
 
-        // Generate code for the callee expression and its value
         ExprBase *calleeExpr = expr->getCallee();
 
-        // Generate code for each argument
         llvm::SmallVector<gil::Value, 4> argValues;
         for (ExprBase *arg : expr->getArgs()) {
             argValues.push_back(visit(arg));
         }
 
         gil::CallInst *callInst = nullptr;
 
-        // If the callee is a reference expression pointing to a FunctionDecl,
-        // emit a direct call
         if (auto *ref = llvm::dyn_cast<RefExpr>(calleeExpr)) {
-            if (FunctionDecl *directCallee
-                = llvm::dyn_cast<FunctionDecl *>(ref->getVariable())) {
+            auto varPU = ref->getVariable();
+            if (varPU && varPU.is<FunctionDecl *>()) {
+                auto *directCallee = varPU.get<FunctionDecl *>();
```

Comment:
just use `dyn_cast_if_present`... but really it should always be present so it should indeed be an assertion failure/crash if it isn't.